### PR TITLE
Data VRF CLI support

### PR DESCRIPTION
**Pull request by mingjunzhang2019**
This is CLI to support data VRF:

- VRF configuration: create/delete
- interfaces (vlan, portchannel, ethernet and loopback) configurations binding to VRF
- show individual VRF (show ip vrf <vrf-name>)

Test outputs:
~~~text
sonic# show ip vrf Vrf_red
sonic# configure terminal
sonic(config)# ip vrf
  management                  Management VRF configuration
  String(Max: 15 characters)  Name of VRF (Max size 15, prefixed by Vrf)

sonic(config)# ip vrf vrf_red
%Error: Invalid VRF name
sonic(config)# ip vrf Vrf_red

sonic# show ip vrf Vrf_red
VRF-NAME            INTERFACES
----------------------------------------------------------------
Vrf_red

sonic(config)# interface Ethernet 4
sonic(conf-if-Ethernet4)# ip vrf forwarding Vrf_red
sonic(conf-if-Ethernet4)# exit
sonic(config)# interface Vlan 120
sonic(conf-if-Vlan120)# ip vrf forwarding Vrf_red
sonic(conf-if-Vlan120)# exit
sonic(config)# interface PortChannel 121
sonic(conf-if-po121)# ip vrf forwarding Vrf_red
sonic(conf-if-po121)# exit
sonic(config)# interface Loopback 2
sonic(conf-if-lo2)# ip vrf forwarding Vrf_red
sonic(conf-if-lo2)# do show ip vrf Vrf_red
VRF-NAME            INTERFACES
----------------------------------------------------------------
Vrf_red             Ethernet4
                    Loopback2
                    PortChannel121
                    Vlan120
sonic(conf-if-lo2)# no ip vrf forwarding Vrf_red
sonic(conf-if-lo2)# do show ip vrf Vrf_red
VRF-NAME            INTERFACES
----------------------------------------------------------------
Vrf_red             Ethernet4
                    PortChannel121
                    Vlan120
~~~

Not supported:

- show all VRF's (show ip vrf) - needs transformer support

### CODE REVIEW

**Review by bsun-sudo — APPROVED** [review]

File: src/CLI/clitree/macro/vrf_macro.xml [hunk 1]
```diff
@@ -28,12 +28,12 @@ limitations under the License.
         >
         <PARAM
           name="vrf-name"
-          help="Name of VRF (Max size 32, prefixed by Vrf_)"
```
Comment (venkatmahalingam) [inline src/CLI/clitree/macro/vrf_macro.xml:31]: VRF name length should be 15 chars, please refer HLD for more information.
https://github.com/Azure/SONiC/pull/435/files

File: src/CLI/renderer/templates/show_vrf.j2 [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
-        {% set vrf_info = json_output['openconfig-network-instance:config'] %}
-        {% if vrf_info['name'] == 'mgmt' %}
+    {% set vrfs = json_output['openconfig-network-instance:network-instance'] %}
+    {% for vrf in vrfs %}
+        {% if vrf['name'] == 'mgmt' %}
             {{-'mgmt'.ljust(20)}}{{'eth0'}}
+        {% else %}
+            {% set intfs = vrf['interfaces']['interface'] %}
+            {% set ns = namespace(vrf_name=vrf['name']) %}
+            {% for intf in intfs %}
```
Comment (venkatmahalingam) [inline src/CLI/renderer/templates/show_vrf.j2]: can you paste the UT logs in this PR?

**Review by venkatmahalingam — CHANGES_REQUESTED** [review]
Please address the comments.

File: src/CLI/renderer/templates/show_vrf.j2 (hunk 2 above)
Comment (mingjunzhang2019) [reply src/CLI/renderer/templates/show_vrf.j2]: Done

File: src/CLI/clitree/macro/vrf_macro.xml (hunk 1 above)
Comment (mingjunzhang2019) [reply src/CLI/clitree/macro/vrf_macro.xml:31]: Changed.

File: src/CLI/actioner/sonic-cli-vrf.py [hunk 3]
```diff
@@ -45,13 +45,48 @@ def invoke_api(func, args=[]):
         keypath = cc.Path('/restconf/data/openconfig-network-instance:network-instances/network-instance={name}', name=args[0])
         return api.delete(keypath)
```
Comment (venkatmahalingam) [inline src/CLI/actioner/sonic-cli-vrf.py:46]: sonic# show ip vrf Vrf_red
%Error: Vrf_red is not exist.
sonic#

We should not throw an error when the VRF does not exist.

Comment (mingjunzhang2019) [reply src/CLI/actioner/sonic-cli-vrf.py:46]: Removed that error message.

**Review by venkatmahalingam — APPROVED** [review]

**Pull request merged**